src/launchpad/artifacts/providers/zip_provider.py

```diff
@@ -8,6 +8,59 @@
 
 logger = get_logger(__name__)
 
+DEFAULT_MAX_FILE_COUNT = 100000
+DEFAULT_MAX_UNCOMPRESSED_SIZE = 10 * 1024 * 1024 * 1024
+
+
+class UnreasonableZipError(ValueError):
```

Comment:
nit: Maybe the traversal logic should have its own error subclass as well?

Reply:
done, thanks!